packages/Button/src/Button.styles.js

```diff
@@ -46,11 +68,18 @@ const disabledTextStyles = css`
 `;
 
 const activeStyles = css`
-  border-color: ${tokens.highlight.active.noBorder.borderColor};
-  box-shadow: ${tokens.highlight.active.noBorder.boxShadow};
+  ${stylers.focusRing.bordered()}
+`;
 
-  &:hover:not([disabled]):not([aria-disabled="true"]) {
-    border-color: ${tokens.highlight.active.noBorder.borderColor};
+const mouseStyles = css`
+  [data-whatinput="mouse"] &:not([data-has-forced-focus="true"]):focus {
+    border-color: ${({ kind }) => borderColors[kind]};
+    box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.1);
+    ${({ kind }) => [Kinds.FLAT, Kinds.MINOR, Kinds.LINK].includes(kind) && "box-shadow: none;"}
```

Comment:
I think i'd rather see the `box-shadow` evaluated in its own function rather than overriding the previous line 77. Otherwise it these two could easily be ordered by accident down the line.

Reply:
Actually decided that the "not disabled" selector would benefit from a helper. It's somewhat specific to a button though, so not sure it should get moved to the `helpers` package:

https://github.com/acl-services/paprika/blob/e9e1b3e76d3e0f848f547304c2e53347508555d8/packages/Button/src/Button.styles.js#L9-L13